plugin/src/main/java/org/opensearch/ml/action/memorycontainer/TransportCreateMemoryContainerAction.java

```diff
@@ -149,20 +149,16 @@ private MLMemoryContainer buildMemoryContainer(MLCreateMemoryContainerInput inpu
             .build();
     }
 
-    private void createMemoryDataIndices(
-        MLMemoryContainer container,
-        User user,
-        ActionListener<String> listener
-    ) {
+    private void createMemoryDataIndices(MLMemoryContainer container, User user, ActionListener<String> listener) {
         String userId = user != null ? user.getName() : "default";
         MemoryConfiguration configuration = container.getConfiguration();
         String indexPrefix = configuration != null ? configuration.getIndexPrefix() : null;
 
         // Convert to lowercase as OpenSearch doesn't support uppercase in index names
-        final String sessionIndexName = indexPrefix.toLowerCase(Locale.ROOT) + "-session";
-        final String shortTermMemoryIndexName = indexPrefix.toLowerCase(Locale.ROOT) + "-short-term-memory";
-        final String longTermMemoryIndexName = indexPrefix.toLowerCase(Locale.ROOT) + "-long-term-memory";
-        final String longTermMemoryHistoryIndexName = configuration.getLongMemoryHistoryIndexName();
+        final String sessionIndexName = configuration.getSessionIndexName().toLowerCase(Locale.ROOT);
```

Comment:
Let's move `toLowerCase` to `configuration.getSessionIndexName` to avoid duplicate this in multiple places?

A better way is converting `indexPrefix` to lowercase in `MemoryConfiguration` class constructor method. So just need to convert to lower case one time, not need to convert lower case for every getXXXIndexName request